ViV-Wallet-api/src/main/java/com/invivoo/vivwallet/api/infrastructure/lynx/LynxConnector.java

```diff
@@ -53,15 +48,21 @@ public List<Action> findActions() {
     public List<Action> getActionsFromActivities(List<Activity> activities) {
         LocalDateTime validityDate = LocalDateTime.now();
         List<Activity> validatedActivities = activities.stream()
-                                                       .filter(activity -> Objects.nonNull(activity.getType()) && StringUtils.isNotBlank(activity.getOwner()))
-                                                       .filter(activity -> Optional.ofNullable(activity.getValueDate()).filter(validityDate::isAfter).isPresent())
-                                                       .filter(activity -> ACTIVITY_HELD_STATUS.equals(activity.getStatus()) && ACTIVITY_IS_OK.equals(activity.getValidity()))
+                                                       .filter(activity -> Objects.nonNull(activity.getType()) && StringUtils.isNotBlank(
```

Comment:
I think you should replace the lambda expression in these filter and extract them into method call, so that the method name give us a better understanding of what is going on.

Reply:
This was automatic refactoring because of my new code style but you are right and I'll do it, it will bie easier to read